api/handlers/service_plan_test.go

```diff
@@ -350,4 +351,46 @@ var _ = Describe("ServicePlan", func() {
 			})
 		})
 	})
+
+	Describe("DELETE /v3/service_plans/{guid}/visibility/{org-guid}", func() {
+		BeforeEach(func() {
+			servicePlanRepo.DeletePlanVisibilityReturns(nil)
+		})
+
+		JustBeforeEach(func() {
+			req, err := http.NewRequestWithContext(ctx, "DELETE", "/v3/service_plans/my-service-plan/visibility/org-guid", nil)
+			Expect(err).NotTo(HaveOccurred())
+
+			routerBuilder.Build().ServeHTTP(rr, req)
+		})
+
+		It("deletes the service plan visibility", func() {
+			Expect(servicePlanRepo.DeletePlanVisibilityCallCount()).To(Equal(1))
+			_, actualAuthInfo, actualMessage := servicePlanRepo.DeletePlanVisibilityArgsForCall(0)
+			Expect(actualAuthInfo).To(Equal(authInfo))
+			Expect(actualMessage.PlanGUID).To(Equal("my-service-plan"))
+			Expect(actualMessage.OrgGUID).To(Equal("org-guid"))
+			Expect(rr).To(HaveHTTPStatus(http.StatusNoContent))
+		})
+
+		When("deleting the visibility fails with not found", func() {
```

Comment:
The handler does not process not found any differently than a generic error, so this test does not add more value then the one below.